ESPHome 2025.2.0 Config with PbHub fails to compile

Hi, thanks for the work on this external component!  I'm using a PbHub in this way:
```
M5CameraF <---> PbHub <--|--> Unit Button
                         |--> Unit PIR
```
The full config for this device:
```
esphome:
  name: main-door-camera
  friendly_name: Main Door Camera

esp32:
  board: esp32dev
  framework:
    type: arduino

# Enable logging
logger:

# Enable Home Assistant API
api:
  encryption:
    key: !secret main_door_camera_api_key

ota:
  - platform: esphome
    password: !secret main_door_camera_ota_password

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password

  # Enable fallback hotspot (captive portal) in case wifi connection fails
  ap:
    ssid: "Main-Door-Camera"
    password: !secret ap_password

  manual_ip:
    static_ip: 192.168.18.58
    gateway: 192.168.18.1
    subnet: 255.255.255.0

captive_portal:
    
external_components:
  - source:
      type: git
      url: https://github.com/Chill-Division/M5Stack-ESPHome/
    components: m5stack_pbhub

i2c:
  id: grove
  sda: 4
  scl: 13

m5stack_pbhub:
  - id: pbhub
    address: 0x61
    sda: 4
    scl: 13

binary_sensor:
  - platform: gpio
    id: motion
    name: Motion
    pin: 
      m5stack_pbhub: pbhub
      number: 31
      mode:
        output: true
    device_class: motion
    publish_initial_state: true
    filters:
      - delayed_off: 20s
    on_press: 
      then:
        - light.turn_on: 
            id: led
            effect: pulse
    on_release: 
      then:
        - light.turn_off: led
  - platform: gpio
    id: button
    name: Button
    pin: 
      m5stack_pbhub: pbhub
      number: 11
      mode:
        input: true
    publish_initial_state: true
    filters:
      - invert:

output:
  - platform: ledc
    pin: 14
    inverted: true
    id: led_output

light:
  - platform: monochromatic
    output:  led_output
    name: LED
    id: led
    effects:
      - pulse:
          name: "Pulse"
          transition_length: 1000ms
          update_interval: 1000ms
          min_brightness: 10%
          max_brightness: 100%

esp32_camera:
  id: camera
  external_clock:
    pin: GPIO27
    frequency: 20MHz
  i2c_pins:
    sda: GPIO22
    scl: GPIO23
  data_pins: [GPIO32, GPIO35, GPIO34, GPIO5, GPIO39, GPIO18, GPIO36, GPIO19]
  vsync_pin: GPIO25
  href_pin: GPIO26
  pixel_clock_pin: GPIO21
  reset_pin: GPIO15
  max_framerate: 2 fps
  jpeg_quality: 20
  brightness: 2

esp32_camera_web_server:
  - port: 8080
    mode: stream
  - port: 8081
    mode: snapshot
```

This successfully worked under ESPHome 2024.12, but upgrading to 2025.2 caused the compile to fail:

```
INFO ESPHome 2025.2.0
INFO Reading configuration /config/esphome/main-door-camera.yaml...
WARNING GPIO5 is a strapping PIN and should only be used for I/O with care.
Attaching external pullup/down resistors to strapping pins can cause unexpected failures.
See https://esphome.io/guides/faq.html#why-am-i-getting-a-warning-about-strapping-pins
WARNING GPIO15 is a strapping PIN and should only be used for I/O with care.
Attaching external pullup/down resistors to strapping pins can cause unexpected failures.
See https://esphome.io/guides/faq.html#why-am-i-getting-a-warning-about-strapping-pins
INFO Generating C++ source...
INFO Compiling app...
Processing main-door-camera (board: esp32dev; framework: arduino; platform: platformio/espressif32@5.4.0)
--------------------------------------------------------------------------------
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
 - toolchain-xtensa-esp32 @ 8.4.0+2021r2-patch5
Dependency Graph
|-- AsyncTCP-esphome @ 2.1.4
|-- WiFi @ 2.0.0
|-- FS @ 2.0.0
|-- Update @ 2.0.0
|-- ESPAsyncWebServer-esphome @ 3.2.2
|-- DNSServer @ 2.0.0
|-- ESPmDNS @ 2.0.0
|-- noise-c @ 0.1.6
|-- Wire @ 2.0.0
Compiling .pioenvs/main-door-camera/src/main.cpp.o
Archiving .pioenvs/main-door-camera/lib01c/libFS.a
Compiling .pioenvs/main-door-camera/libbc6/Update/Updater.cpp.o
Compiling .pioenvs/main-door-camera/lib22e/ESPAsyncWebServer-esphome/AsyncEventSource.cpp.o
Compiling .pioenvs/main-door-camera/lib22e/ESPAsyncWebServer-esphome/AsyncWebSocket.cpp.o
src/main.cpp: In function 'void setup()':
src/main.cpp:317:69: error: invalid new-expression of abstract class type 'esphome::m5stack_pbhub::PBHUBGPIOPin'
   m5stack_pbhub_pbhubgpiopin_id_2 = new m5stack_pbhub::PBHUBGPIOPin();
                                                                     ^
In file included from src/esphome.h:51,
                 from src/main.cpp:3:
src/esphome/components/m5stack_pbhub/m5stack_pbhub.h:47:7: note:   because the following virtual functions are pure within 'esphome::m5stack_pbhub::PBHUBGPIOPin':
 class PBHUBGPIOPin : public GPIOPin {
       ^~~~~~~~~~~~
In file included from src/esphome/core/hal.h:4,
                 from src/esphome/components/light/light_transformer.h:4,
                 from src/esphome/components/light/light_state.h:11,
                 from src/esphome/core/controller.h:11,
                 from src/esphome/components/api/api_server.h:11,
                 from src/esphome/components/api/api_connection.h:8,
                 from src/esphome.h:3,
                 from src/main.cpp:3:
src/esphome/core/gpio.h:61:23: note: 	'virtual esphome::gpio::Flags esphome::GPIOPin::get_flags() const'
   virtual gpio::Flags get_flags() const = 0;
                       ^~~~~~~~~
src/main.cpp:510:67: error: invalid new-expression of abstract class type 'esphome::m5stack_pbhub::PBHUBGPIOPin'
   m5stack_pbhub_pbhubgpiopin_id = new m5stack_pbhub::PBHUBGPIOPin();
                                                                   ^
Archiving .pioenvs/main-door-camera/libbc6/libUpdate.a
Compiling .pioenvs/main-door-camera/lib22e/ESPAsyncWebServer-esphome/WebAuthentication.cpp.o
*** [.pioenvs/main-door-camera/src/main.cpp.o] Error 1
========================== [FAILED] Took 9.09 seconds ==========================
```
For now, the device still works with the old config but I cannot upgrade it to 2025.2.  Please let me know if I can get more info to help with this issue.

--- a/components/m5stack_pbhub/m5stack_pbhub.h
+++ b/components/m5stack_pbhub/m5stack_pbhub.h
@@ -57,6 +57,8 @@ class PBHUBGPIOPin : public GPIOPin {
   void set_inverted(bool inverted) { inverted_ = inverted; }
   void set_flags(gpio::Flags flags) { flags_ = flags; }
 
+  gpio::Flags get_flags() const override { return this->flags_; }
+
  protected:
   M5StackPBHUBComponent *parent_;
   uint8_t pin_;

